simulator.go

```diff
@@ -202,13 +204,46 @@ type simulatorAPIHandler struct {
 	overrides []string
 	autoID    uint32
 
-	runner *docker.Container
-	nodes  map[string]*docker.Container
+	runner       *docker.Container
+	nodes        map[string]*docker.Container
+	nodesTimeout map[string]time.Time
 
 	result *simulationResult
 	lock   sync.RWMutex
 }
 
+// CheckTimeout is a goroutine that checks if the timeout has passed and stops
+// container if it has.
+func (h *simulatorAPIHandler) CheckTimeout() {
+	for {
+		for id, c := range h.nodes {
+			h.lock.Lock()
+			if !c.State.Running || (time.Now().After(h.nodesTimeout[id])) {
+				h.terminateContainer(id, nil)
+			}
+			h.lock.Unlock()
+		}
+		time.Sleep(timeoutCheckDuration)
```

Comment:
Nitpick, it's good practice to have a close channel, which can be fired when the program is shutting down. Doesn't really matter though

Reply:
I moved the locks outside the loop. I'm not sure if they will take up too much time when it loops through all the containers to check for the timeout. If it does, we can likely do something with RLock and such, but yeah, it's complicated. I usually try to avoid mutex's for the most part (the share memory by communicating thing). Just let me know if you want me to change something.

Reply:
I've seen a close channel when the function spins off it's own goroutine so that a user of the library knows there's a goroutine that was started. I could modify the function to do that, but l'm not sure if it's necessary. If you want that, I'll make that change.

Also, the goroutine that's started just after this one (line 189) is ignoring the error from http.Serve. Not sure if you knew about that or if it's intentional, but I thought I'd mention it since I saw it.

Reply:
Oh, and sorry for being a bit slow on these changes. I have a cold this week and it's just killing me.